File: backend/models/transaction.go
package models

type Transaction struct {
	Id int `json:"id"`
	Title string `json:"title"`
	Description string `json:"description"`
	Amount int64 `json:"amount"`
	Date string `json:"date"`
	Bank string `json:"bank"`
	Category string `json:"category"`
	Type string `json:"type"`
}

File: android/app/src/main/java/com/keef/keef/model/Transaction.kt
package com.keef.keef.model

data class Transaction(
    val id: Int = 0,
    val title: String,
    val description: String? = null,
    val amount: Long,
    val transactionType: TransactionType,
    val datetime: String,
    val bank: Bank,
    val category: Category,
)

enum class Bank(val title: String) {
    Pasargad("پاسارگاد"), Mellat("ملت"), Blu("بلو"), Wepod("ویپاد"), MehrIran("مهرایران");

    override fun toString(): String = this.title
}

enum class TransactionType(val prefix: String) {
    Deposit("به"), Withdraw("از");

    override fun toString(): String {
        return if (this == Deposit) "واریز" else "برداشت"
    }
}
